### PR TITLE
fix: add `start_time` to `ProcessUID` on `StorageLock`

### PR DESCRIPTION
**Pull request by joshieDo**
Makes the identifier more unique by adding `start_time`. 

Fixes the following kind of issue:

1. `reth node` with PID 9 locks storage.
2. `reth node` stops non-gracefully leaving the lockfiles behind.
3. `firefox` starts with PID 9.
4. `reth node` attempts to start but realizes that PID 9 is executing, and thus, does not acquire the lock.

### CODE REVIEW

**Review by shekhirin — APPROVED** [review]

**Review by DaniPopes — APPROVED** [review]

**Review by mattsse — APPROVED** [review]
I think this should work

**Review by gakonst — APPROVED** [review]

File: crates/storage/db/src/lockfile.rs [hunk 1]
```diff
@@ -28,15 +30,24 @@ impl StorageLock {
...
+            // Too expensive for ef-tests to write/read lock to/from disk.
+            Ok(Self(Arc::new(StorageLockInner { file_path })))
         }
 
-        Ok(Self(Arc::new(StorageLockInner::new(path)?)))
+        #[cfg(not(feature = "disable_lock"))]
+        {
+            if let Some(process_lock) = ProcessUID::parse(&file_path)? {
+                if process_lock.pid != (process::id() as usize) && process_lock.is_active() {
+                    return Err(StorageLockError::Taken(process_lock.pid))
```
Comment (mattsse) [inline crates/storage/db/src/lockfile.rs:45]: could we add an error trace here?

File: crates/storage/db/Cargo.toml [hunk 2]
```diff
@@ -78,6 +78,7 @@ mdbx = ["reth-libmdbx"]
 bench = []
 arbitrary = ["reth-primitives/arbitrary", "reth-db-api/arbitrary"]
 optimism = []
+disable_lock = []
```
Comment (mattsse) [inline crates/storage/db/Cargo.toml:81]:
```suggestion
disable-lock = []
```

Comment (mattsse) [inline crates/storage/db/Cargo.toml:81]: if this feature enabled do we then need sysinfo dep?

opting out is a lot harder than opting in, but in this case we expect that this is only disabled for testing, so imo this is fine

**Review by mattsse — APPROVED** [review]
pedantic suggestions

**Pull request merged**